### PR TITLE
Data layer: Set auth token for oAuth apps

### PR DESCRIPTION
**Pull request by mmtr**
#### Changes proposed in this Pull Request

Sets the authentication token for data-layer requests originated on oAuth apps as done on the _undocumented_ wpcom.js:

https://github.com/Automattic/wp-calypso/blob/6a593a0d88df8d7622d82f31ce3503d3264f087c/client/lib/wpcom-undocumented/index.js#L38-L42

https://github.com/Automattic/wp-calypso/blob/6a593a0d88df8d7622d82f31ce3503d3264f087c/client/lib/wp/browser.js#L22-L23

#### Testing instructions

This needs to be tested on the desktop app, so make sure you have the dev environment [installed and running](https://github.com/Automattic/wp-desktop/blob/develop/docs/install.md).

- In the desktop repo, point the calypso submodule to this branch: `cd calypso && git fetch && git checkout update/data-layer-auth-token && cd ..`.
- Generate a new build: `make build`.
- Open the built app in the `release/<YOUR_OS>` folder (i.e. `release/mac/WordPress.com.app`).
- Switch to a site with Customer Home enabled and go to My Home.
- Select "Email download link" on the "Go Mobile" card.
- Make sure you get an email with a link to download the app.

Fixes https://github.com/Automattic/wp-desktop/issues/777

### CODE REVIEW

File: client/state/data-layer/wpcom-http/index.js [hunk 1]
```diff
@@ -69,7 +73,7 @@ export const queueRequest = ( processOutbound, processInbound ) => ( { dispatch
 					return null;
 				}
 
-				return !! nextError
+				return nextError
```
Comment (mmtr) [inline client/state/data-layer/wpcom-http/index.js]: _`eslint` fix_

**Comment by mmtr** [comment]
Uhm, seems this does not help fixing https://github.com/Automattic/wp-desktop/issues/777. Folks are welcome to jump in and push changes to this branch.

I verified the request is done with an authorization header that matches the token returned by `oauthToken.getToken()` but the API still refuses it:

Screen Shot 2020-02-21 at 13 40 27

```
{"error":"invalid_token","message":"Bearer token not for specified client_id"}
```

**Comment by gwwar** [comment]
Hmm, we'll need to probably compare the expected full undocumented request (in an earlier version) vs what we have now. The me -> get apps page for the magic link also uses the same endpoint.

Remember to not paste recorded API responses in GH

**Comment by dmsnell** [comment]
One thing I might challenge us to ask here is if this would be better off as a new _pipeline_ process which adds the authentication to the requests.

The nature of the iframe-proxy-request is that it hides the specific authentication information from the general description of the request and here we're putting them back together.

This looks fine to me, especially since it's a pretty core aspect of the requests and isn't likely to proliferate. That being said, maybe it's still worth comparing against a pipeline approach.

```ts
const token = oauthToken.getToken();

const addOAuthToken = outboundData => {
	const { nextRequest } = outboundData;

	if ( nextRequest.authToken && nextReqeust.token ) {
		return outboundData;
	}

	const withAuth = { ...nextRequest, authToken: token, token };
	return { ...outboundData, nextRequest: withToken };
}
```

```ts
const outboundChain = [
	removeDuplicateGets,
	config.isEnabled('oauth') && addOAuthToken,
].filter(Boolean);
```

**Comment by mmtr** [comment]
> Hmm, we'll need to probably compare the expected full undocumented request (in an earlier version) vs what we have now. The me -> get apps page for the magic link also uses the same endpoint.
> 
> Remember to not paste recorded API responses in GH

Seems it wasn't working previously. I tested on WP.com desktop 4.7.0 where the email download link is requested using `wpcom.js` undocumented and the API is still refusing it with the same bearer error.

I think we have a more generic issue in the desktop app that is unrelated to the data layer.

**Comment by mmtr** [comment]
Confirmed that the issue is completely unrelated to the data later or the undocumented `wpcom.js` library, and it's part of a bigger issue. See https://github.com/Automattic/wp-desktop/issues/777#issuecomment-590390781 for further details.

**Pull request closed without merging**